text/0063-attestation-grpc-apis.md

```diff
@@ -0,0 +1,316 @@
+        Client-->>Enclave: Secure message
+        Enclave-->>Client: Secure message
+    end
+```
+
+The purpose of this MCIP is to better define the data format of the attestation
+evidence that enclaves provide to clients.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+There are two common ways of attesting with an enclave:
+
+1. [Client to enclave](#1-client-to-enclave)
+2. [Enclave to enclave](#2-enclave-to-enclave)
+
+The provided services support these two ways:
+
+```prost
+service AttestService {
+    rpc Attest(AttestRequest) returns (AttestResponse);
+    rpc MutualAttest(MutualAttestRequest) returns (MutualAttestResponse);
+}
+```
+
+The regular `Attest` endpoint is for client to enclave attestation. The
+`MutualAttest` endpoint is for enclave to enclave attestation.
+
+## 1. Client to Enclave
+
+Client to enclave is for usages like a mobile device or a wallet communicating
+with an enclave.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    Client->>+Enclave: AttestRequest
+    Enclave->>+Client: AttestResponse
+    Client->>Client: Verify AttestResponse contents
+    loop Ongoing Session
+        Client-->>Enclave: AttestedMessage
+        Enclave-->>Client: AttestedMessge
+    end
+```
+
+## 2. Enclave to Enclave
+
+Enclave to enclave attestation is for enclaves on disparate machines
+communicating with each other. A common use case is for communication between
+concensus nodes.
+
+Enclave to encalve attestation differs from
+[client to enclave](#1-client-to-enclave) in that the requester will provide its
+own attestation evidence when requesting attestation evidence from the other
+enclave. The enclave recieving the request will verify the evidence of the
+requesting enclave before responding.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    Enclave_1->>+Enclave_2: MutualAttestRequest
+    Enclave_2->>Enclave_2: Verify MutualAttestRequest contents
+    Enclave_2->>+Enclave_1: MutualAttestResponse
+    Enclave_1->>Enclave_1: Verify MutualAttestResponse contents
+    loop Ongoing Session
+        Enclave_1-->>Enclave_2: AttestedMessage
+        Enclave_2-->>Enclave_1: AttestedMessage
+    end
+```
+
+## Message Types
+
+### `AttestRequest`
+
+Used to initiate client to enclave communication.
+
+The `noise` field contains
+[noise protocol](http://www.noiseprotocol.org/) handshake data.
+
+```prot
+message AttestRequest {
+    bytes noise = 1;
+}
+```
+
+### `AttestResponse`
+
+Contains the noise protocol handshake data and the attestation evidence that
+needs to be verified by the recipient
+
+```proto
+message AttestResponse {
+    bytes noise = 1;
+    AttestationEvidence evidence = 2;
+}
+```
+
+### `AttestationEvidence`
+
+A [`oneof`](https://protobuf.dev/programming-guides/proto3/#oneof) is used to
+allow for new evidence types to be provided in the future.
+
+The name `quote3` was chosen as that is how the binary format is refered to in
+the Intel SDK.
```

Comment:
In writing this it re-emphasized the DCAP == Quote3 naming.
Up for suggestions here.
Maybe this should have a section saying
DCAP is the enclave -> service protocol
while Quote3 is the evidence format

Reply:
Yeah, it's probably worth defining the difference somewhere up at the top.

Reply:
Add note to each message to help clarify